File: apps/mobile/modules/host-environment/android/src/main/java/expo/modules/hostenvironment/HostEnvironmentModule.kt
package expo.modules.hostenvironment

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * RN-facing surface for the host-selected backend environment. Storage lives
 * in [HostEnvironment]; this module only bridges the JS read. Returns null
 * when no host has published an environment (standalone builds, dev).
 */
class HostEnvironmentModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("HostEnvironment")

    Function("getEnvironment") {
      HostEnvironment.current
    }
  }
}

File: apps/mobile/modules/host-environment/ios/HostEnvironmentModule.swift
import ExpoModulesCore

/// RN-facing surface for the host-selected backend environment. Storage lives
/// in `HostEnvironmentRegistry`; this module only bridges the JS read. Returns
/// nil when no host has published an environment (standalone builds, dev).
public class HostEnvironmentModule: Module {
  public func definition() -> ModuleDefinition {
    Name("HostEnvironment")

    Function("getEnvironment") { () -> String? in
      return HostEnvironmentRegistry.shared.current
    }
  }
}
